dt-posts/dt-posts.php

```diff
@@ -676,7 +676,7 @@ public static function add_post_comment( string $post_type, int $post_id, string
         return $created_comment_id;
     }
 
-    public static function update_post_comment( int $comment_id, string $comment_content, bool $check_permissions = true ){
+    public static function update_post_comment( int $comment_id, string $comment_content, string $comment_type = "comment", bool $check_permissions = true ){
```

Comment:
@micahmills this breaks anywhere that is using this function with the $check_permissions param.

Reply:
do you have a suggestion for how to fix this?

Reply:
In php add the args at the end of the list, so it keeps the same order for all functions.
here i'd suggest we future proof it and format it like this:
```
    public static function update_post_comment( int $comment_id, string $comment_content, bool $check_permissions = true, $args = [] ){
        $comment = get_comment( $comment_id );
        if ( $check_permissions && ( ( isset( $comment->user_id ) && $comment->user_id != get_current_user_id() ) || !self::can_update( get_post_type( $comment->comment_post_ID ), $comment->comment_post_ID ?? 0 ) ) ) {
            return new WP_Error( __FUNCTION__, "You don't have permission to edit this comment", [ 'status' => 403 ] );
        }
        if ( !$comment ){
            return new WP_Error( __FUNCTION__, "No comment found with id: " . $comment_id, [ 'status' => 403 ] );
        }
        $comment = [
            "comment_content" => $comment_content,
            "comment_ID" => $comment_id,
        ];
        if ( isset( $args["comment_type"] ) ) {
            $comment["comment_type"] = $args["comment_type"];
        }
        return wp_update_comment( $comment );
    }
```

Reply:
Yeah that makes sense, here I thought it was something difficult and was thinking too hard about it.